.travis.yml

```diff
@@ -1,43 +1,44 @@
 sudo: required
 language: generic
 
+dist: bionic
+
 services:
 - docker
 
 before_script:
-- curl -sSLf https://get.docker.com | sed s/sleep\ 20/sleep\ 0/g | sudo -E sh
+  - sudo rm -rf /var/lib/apt/lists/*
+  - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+  - sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) edge"
+  - sudo apt-get update -qy
+  - sudo apt-get -y -o Dpkg::Options::="--force-confnew" install docker-ce
+  - mkdir -vp ~/.docker/cli-plugins/
+  - curl --silent -L "https://github.com/docker/buildx/releases/download/v0.3.0/buildx-v0.3.0.linux-amd64" > ~/.docker/cli-plugins/docker-buildx
+  - chmod a+x ~/.docker/cli-plugins/docker-buildx
+  - sudo systemctl start docker
+  - ./ci/install-buildx.sh
+  - docker buildx version
 
 script:
+  - make docker || travis_terminate 1;
   - make build
+  - make build_redist
   - make build_samples
   - ./test/version.sh ./faas-cli
-  - make build_redist
 
 after_success:
-    - if [ ! -z "$TRAVIS_TAG" ] ; then
-        if [ -z "$DOCKER_NS" ] ; then
-          export DOCKER_NS=openfaas;
-        fi;
-
-        docker tag $DOCKER_NS/faas-cli:latest-dev $DOCKER_NS/faas-cli:$TRAVIS_TAG;
-        docker tag $DOCKER_NS/faas-cli:latest-dev $DOCKER_NS/faas-cli:latest;
-        docker tag $DOCKER_NS/faas-cli:latest-dev-root $DOCKER_NS/faas-cli:${TRAVIS_TAG}-root;
-        docker tag $DOCKER_NS/faas-cli:latest-dev-root $DOCKER_NS/faas-cli:latest-root;
-        echo $DOCKER_PASSWORD | docker login -u=$DOCKER_USERNAME --password-stdin;
-        docker push $DOCKER_NS/faas-cli:$TRAVIS_TAG;
-        docker push $DOCKER_NS/faas-cli:latest;
-        docker push $DOCKER_NS/faas-cli:${TRAVIS_TAG}-root;
-        docker push $DOCKER_NS/faas-cli:latest-root;
-
-        docker tag $DOCKER_NS/faas-cli:latest-dev quay.io/$DOCKER_NS/faas-cli:$TRAVIS_TAG;
-        docker tag $DOCKER_NS/faas-cli:latest-dev quay.io/$DOCKER_NS/faas-cli:latest;
-        docker tag $DOCKER_NS/faas-cli:latest-dev-root quay.io/$DOCKER_NS/faas-cli:${TRAVIS_TAG}latest-root;
-        docker tag $DOCKER_NS/faas-cli:latest-dev-root quay.io/$DOCKER_NS/faas-cli:latest-root;
-        echo $QUAY_PASSWORD | docker login -u=$QUAY_USERNAME --password-stdin quay.io;
-        docker push quay.io/$DOCKER_NS/faas-cli:$TRAVIS_TAG;
-        docker push quay.io/$DOCKER_NS/faas-cli:latest;
-        docker push quay.io/$DOCKER_NS/faas-cli:${TRAVIS_TAG}-root;
-        docker push quay.io/$DOCKER_NS/faas-cli:latest-root;
+  - make docker-login
+  - make docker-login-ghcr
+  - if [ ! -z "$TRAVIS_TAG" ] ; then
```

Comment:
We shouldn't need this, or it can move to the Makefile

Reply:
See how I did it for [inlets/inlets](https://github.com/inlets/inlets/blob/master/Makefile)

Reply:
i wasn't sure what you wanted to do with the quey stuff, so i left it alone

Reply:
Sure. Can we do it like the inlets/inlets repo?

Basically build again with another tag, it just uses the cache.

Let's switch to GHCR for this repo from Quay.io, I just checked that nobody has downloaded it from there all year.

Reply:
(FYI I've added GHCR credentials and I've deleted the Quay.io repo for faas-cli)